### ISSUE TITLE
[a11y]: Toolkit: Focus is going to Parent page even while Switch module menu/list is opened

### ISSUE THREAD

**Issue by ErickRenteria** [carbon-design-system/carbon#16858]
### Package

carbon-components

### Browser

Firefox

### Operating System

Windows

### Package version

10.21.0

### Automated testing tool and ruleset

n/a

### Assistive technology

Keyboard

## Environment

Platform: Desktop
Violation Type: Violation
Operating system: Windows
Browser: Firefox
Section: Toolkit
URL: https://www.ibm.com/able/toolkit

WCAG 2.1 violation: 2.4.3 Focus Order 
Assistive technology used to verify: Keyboard

## Detailed description
Tab focus is going to Parent page while Switch module menu/list is opened, where Focus should not go out of the Menu with out the menu getting closed.
Note: This issue is occurring in 200% and Reflow aswell.

## Additional information
Expected: Recommend to Maintain the Focus with in the Switch module menu else once Tab focus exits Menu, it should get closed.
Impact Statement: When focus is not set on a logical point in the page after user action, this confuses the user and it can be tedious to navigate back to the relevant element in the page. this impacts the Keyboard only users and also people with cognitive disability.

- Screenshots or code
 
![image](https://media.github.ibm.com/user/403425/files/079e3580-1807-11ed-9aaa-1ebd18988757)

### WCAG 2.1 Violation

2.4.3 Focus Order 

### Reproduction/example

N/a

### Steps to reproduce

Tab focus is going to Parent page while Switch module menu/list is opened, where Focus should not go out of the Menu with out the menu getting closed.
Note: This issue is occurring in 200% and Reflow as-well.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
v10.x has reached end of support, but this is still present in v11

**Comment by warrenmblood** [comment]
The `HeaderPanel` component has a prop `onHeaderPanelFocus` that allows you to set a callback function to close the panel when it loses focus. See UI Shell Overview [docs](https://react.carbondesignsystem.com/?path=/docs/components-ui-shell-header--overview) under "Component API"

In this [storybook example](https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-actions-and-switcher) for UI Shell with Switcher, the switcher menu closes on focus loss because the `Switcher` component is placed inside of a `HeaderPanel` with the prop `onHeaderPanelFocus` passed in.

I'm not sure if the Toolkit [example](https://www.ibm.com/able/toolkit) referenced here is following the UI Shell [usage guidelines](https://carbondesignsystem.com/components/UI-shell-right-panel/usage/), which places the switcher menu inside a right panel. If the switcher is implemented like the storybook example/usage guidelines, I think this a11y issue would be avoided without any changes to the UI Shell component.
